rs/embedders/src/wasm_utils/validation.rs

```diff
@@ -1048,6 +1050,29 @@ fn validate_function_section(
     Ok(())
 }
 
+// Checks that the initial size of the wasm (heap) memory is not larger than
+// the allowed maximum size. This is only needed for Wasm64, because in Wasm32 this
+// is checked by Wasmtime.
+fn validate_initial_wasm_memory_size(
+    module: &Module,
+    max_wasm_memory_size_in_bytes: NumBytes,
+) -> Result<(), WasmValidationError> {
+    if let Some(memory) = module.memories.first() {
```

Comment:
How about we loop over all the memories instead of taking the first one? Just in case we support multiple memories at some point in the future.

Reply:
sure, done now